modules/alia-component/src/qbits/alia/component.clj

```diff
@@ -0,0 +1,131 @@
+(ns qbits.alia.component
+  "This namespace provides a way to "
+  (:require [com.stuartsierra.component :as component]
+            [clojure.spec.alpha         :as s]
+            [qbits.alia                 :as alia]))
+
+(defn illegal!
+  [& msgs]
+  (throw (IllegalArgumentException. ^String (reduce str "" msgs))))
+
+(defn ->cluster
+  "Validate cluster configuration, then build a `Cluster` instance
+   out of it, thanks to alia."
+  [config]
+  (when-not (s/valid? ::alia/cluster-options config)
+    (illegal! (s/explain-str ::alia/cluster-options config)))
+  (alia/cluster config))
+
+(defrecord CassandraRegistry [cluster-opts query-opts configs clusters]
+  component/Lifecycle
+  (start [this]
+    (let [clusters (reduce-kv #(assoc %1 %2 (->cluster %3)) {} configs)]
```

Comment:
Do we really want to allow clusters (plural) per component? I have the feeling that having a single component per cluster might be easier to reason about, then the component/system can manage the composition part if necessary. Feel free to prove me wrong, I personally never had to connect to multiple clusters from the same app so my experience is limited here.

Reply:
The idea here was that apps might need to talk to several clusters (in our case: a cross-zone one, and local ones). 

Moreover, logical components might want their own session. The logic behind the component was to hide all this handling.

You're right that most consumers would likely have a single cluster and that config thus becomes a bit complex for the common case, let me see if I can come up with a simpler approach for this case.

Reply:
I guess a solution that allows both could be good. A single cluster component and a multi cluster component build on the single one.

Reply:
I'll work on that.